src/components/AutoComplete/AsyncAutoComplete.test.js

```diff
@@ -0,0 +1,27 @@
+import React from 'react';
+import { shallow } from 'enzyme';
+import AsyncAutoComplete from './AsyncAutoComplete';
+
+test('AsyncAutoComplete is working !!', () => {
+  const optionsArray = ['One', 'Two', 'Three'];
+  const timeout = 500;
+  const handleSearch = callback => callback(setTimeout(optionsArray, timeout));
```

Comment:
This looks to be the source of the propTypes error. Since `setTimeout` returns a number for the time id, we are calling callback with a number.  This should be written as
```jsx
const handleSearch = callback => {
    setTimeout(() => {
      callback(optionsArray);
    }, timeout);
  };
```
to avoid this. I also noticed that since this test is using set Timeouts the assertions below actually do not run.  There are two ways to fix this.  The easiest would be to use the `done` callback provided.
```jsx
test('AsyncAutoComplete is working !!', done => {
...rest of test
setTimeout(() => {
  component.update();
  expect(component.state().options).toEqual(optionsArray);
  expect(component.state().isLoading).toEqual(false);
  done();
+  }, timeout + 1000);
```
The other option is to wrap the whole test in a Promise and return it.  Then you can call the `resolve` method in the same spot as done in the example above.  I would lean towards using the `done` callback.

Reply:
Great, It worked :)